File: sdks/php/src/CreditNotes/Requests/ListCreditNotesRequest.php
<?php

namespace NovaBilling\CreditNotes\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\CreditNotes\Types\ListCreditNotesRequestStatus;

class ListCreditNotesRequest extends JsonSerializableType
{
    /**
     * @var ?string $customerId
     */
    public ?string $customerId;

    /**
     * @var ?string $invoiceId
     */
    public ?string $invoiceId;

    /**
     * @var ?value-of<ListCreditNotesRequestStatus> $status
     */
    public ?string $status;

    /**
     * @var ?float $page
     */
    public ?float $page;

    /**
     * @var ?float $limit
     */
    public ?float $limit;

    /**
     * @param array{
     *   customerId?: ?string,
     *   invoiceId?: ?string,
     *   status?: ?value-of<ListCreditNotesRequestStatus>,
     *   page?: ?float,
     *   limit?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->customerId = $values['customerId'] ?? null;
        $this->invoiceId = $values['invoiceId'] ?? null;
        $this->status = $values['status'] ?? null;
        $this->page = $values['page'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/CreditNotes/Requests/ListCreditNotesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListCreditNotesRequest
{
    [JsonIgnore]
    public string? CustomerId { get; set; }

    [JsonIgnore]
    public string? InvoiceId { get; set; }

    [JsonIgnore]
    public ListCreditNotesRequestStatus? Status { get; set; }

    [JsonIgnore]
    public double? Page { get; set; }

    [JsonIgnore]
    public double? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
